chapters/en-UK/xctest/types_of_testing.md

```diff
@@ -1,21 +1,23 @@
 # Types of Testing
 
-A test is a way of verifying code, and making your assumptions explicit. When you make a change somewhere, tests exist to show the implicit connections between objects.
+A test is a way of verifying code, and making your assumptions explicit. Tests exist to show connections between objects. When you make a change somewhere, tests reveal implicit connections.
 
-In the ruby world the tests are basically considered the application's documentation. This is one of the first port of calls for understanding how systems work after reading the overview/README.
+In the Ruby world the tests are basically considered the application's documentation. This is one of the first port of calls for understanding how systems work after reading the overview/README. It's quite similar with node. Both languages are interpreted languages where even variable name typos only show themselves in runtime errors. Without a compiler, you can catch this stuff only with rigorous testing.
 
-In Cocoa, this is less of a problem as we can rely on the tooling more. Given a compiler, static type systems, and a well integrated static analyser - you can know if your code is going to break someone else's very quickly. However, this is only a lint, the code compiling doesn't mean that it will still correctly show a view controller when you tap a button.
+In Cocoa, this is less of a problem as we can rely on the tooling more. Given a compiler, static type systems, and a well integrated static analyser -- you can know if your code is going to break someone else's very quickly. However, this is only a lint, the code compiling doesn't mean that it will still correctly show a view controller when you tap a button. In other words, the behaviour of your app can still be wrong.
 
-That is what testing is for. I will cover three major topics in this book - there are many, many other types of testing patterns ( for example, last week I was introduced to Model Testing. ) However, These are the dominent patterns that exist within the Cocoa world, and I have experience with them. Otherwise, I'd just be copy & pasting from Wikipedia.
+That is what testing is for. I will cover three major topics in this book. There are many, many other types of testing patterns -- for example, last week I was introduced to Model Testing. However, these three topics are the dominent patterns that exist within the Cocoa world, and I have experience with them. Otherwise, I'd just be copy & pasting from Wikipedia.
```

Comment:
"Usefulness" seems to be not a good fit as you can transform BDD-style tests into xUnit tests. They're more readable and all, I admit. But there's no _need_ to use them in order to test an app properly. Will keep this in mind, though.